internal/bminventory/inventory.go

```diff
@@ -4777,8 +4883,430 @@ func (b *bareMetalInventory) AddOpenshiftVersion(ctx context.Context, ocpRelease
+		}
+	}
+
+	if params.InfraenvCreateParams.AdditionalNtpSources == nil {
+		params.InfraenvCreateParams.AdditionalNtpSources = &b.Config.DefaultNTPSource
+	} else {
+		ntpSource := swag.StringValue(params.InfraenvCreateParams.AdditionalNtpSources)
+
+		if ntpSource != "" && !validations.ValidateAdditionalNTPSource(ntpSource) {
+			err = errors.Errorf("Invalid NTP source: %s", ntpSource)
+			return nil, common.NewApiError(http.StatusBadRequest, err)
+		}
+	}
+
+	openshiftVersion, err := b.versionsHandler.GetVersion(swag.StringValue(params.InfraenvCreateParams.OpenshiftVersion))
+	if err != nil {
+		err = errors.Errorf("Openshift version %s is not supported",
+			swag.StringValue(params.InfraenvCreateParams.OpenshiftVersion))
+		return nil, common.NewApiError(http.StatusBadRequest, err)
+	}
+
+	if params.InfraenvCreateParams.SSHAuthorizedKey != nil && *params.InfraenvCreateParams.SSHAuthorizedKey != "" {
+		if err = validations.ValidateSSHPublicKey(*params.InfraenvCreateParams.SSHAuthorizedKey); err != nil {
+			err = errors.Errorf("SSH key is not valid")
+			return nil, common.NewApiError(http.StatusBadRequest, err)
+		}
+	}
+
+	if params.InfraenvCreateParams.StaticNetworkConfig != nil {
+		if err = b.staticNetworkConfig.ValidateStaticConfigParams(params.InfraenvCreateParams.StaticNetworkConfig); err != nil {
+			return nil, common.NewApiError(http.StatusBadRequest, err)
+		}
+	}
+
+	// set the default value for REST API case, in case it was not provided in the request
+	if params.InfraenvCreateParams.ImageType == "" {
+		params.InfraenvCreateParams.ImageType = models.ImageType(b.Config.ISOImageType)
+	}
+
+	if params.InfraenvCreateParams.ClusterID != nil {
+		if _, err = common.GetClusterFromDB(b.db, *params.InfraenvCreateParams.ClusterID, common.SkipEagerLoading); err != nil {
+			err = errors.Errorf("Cluster ID %s does not exists",
+				params.InfraenvCreateParams.ClusterID.String())
+			return nil, common.NewApiError(http.StatusBadRequest, err)
+		}
+	}
+
+	if kubeKey == nil {
+		kubeKey = &types.NamespacedName{}
+	}
+
+	infraEnv := common.InfraEnv{
+		Generated: false,
+		InfraEnv: models.InfraEnv{
+			ID:                     id,
+			Href:                   url.String(),
+			Kind:                   models.InfraEnvKindInfraEnv,
+			Name:                   swag.StringValue(params.InfraenvCreateParams.Name),
+			OpenshiftVersion:       *openshiftVersion.ReleaseVersion,
+			IgnitionConfigOverride: params.InfraenvCreateParams.IgnitionConfigOverride,
+			StaticNetworkConfig:    b.staticNetworkConfig.FormatStaticNetworkConfigForDB(params.InfraenvCreateParams.StaticNetworkConfig),
+			Type:                   params.InfraenvCreateParams.ImageType,
+			UpdatedAt:              strfmt.DateTime{},
+			AdditionalNtpSources:   swag.StringValue(params.InfraenvCreateParams.AdditionalNtpSources),
+			SSHAuthorizedKey:       swag.StringValue(params.InfraenvCreateParams.SSHAuthorizedKey),
+		},
+		KubeKeyNamespace: kubeKey.Namespace,
```

Comment:
do we allow creating another InfraEnv with the same kubeKey?  Mayve we should check if it already exists and return an error in case it does? @avishayt ?

Reply:
In case we are using REST API (Cloud) then the ID is used for uniqueness and we don't care about the name, in the case of K8S API I think that kube-api will validate it first when trying to create the CR in the ETCD and will fail it before the controller will have a chance to try and reconcile it